Split/Network/Streaming/SseNotifications.swift

```diff
@@ -139,11 +150,31 @@ struct OccupancyNotification: NotificationTypeField {
     enum CodingKeys: String, CodingKey {
         case metrics
     }
+
+    var isControlPriChannel: Bool {
+        return channel?.contains(kControlPriToken) ?? false
+    }
+
+    var isControlSecChannel: Bool {
+        return channel?.contains(kControlSecToken) ?? false
+    }
 }
 
 /// Indicates a streaming error related
-struct StreamingError {
+struct StreamingError: NotificationTypeField {
+    var type: NotificationType {
+        return .sseError
+    }
+
     let message: String
     let code: Int
```

Comment:
I am not completely sure how the HTTP client will handle it, but maybe you should consider `message`, `code`, and `statusCode` to be optionals to consider network errors that are not HTTP error, i.e., errors without status code and (ably) code, such as INTERNET DISCONNECTED, etc.

Reply:
Not sure if I get the comment... These messages are received through streaming channel while connected and they are in that json format. They're not related to http.